plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/eap/OpenShiftEapDeployPathController.java

```diff
@@ -36,7 +36,7 @@ protected String getOutputNameFromSettings(IServerAttributes server, IModule mod
 			String ret = super.getOutputNameFromSettings(server, module);
 			if (ret == null && module.equals(findProjectModule(server))) {
 				String suffix = ServerModelUtilities.getDefaultSuffixForModule(module);
-				ret = "ROOT" + suffix;
+				ret = module.getName() + suffix;
```

Comment:
That would probably fix for EAP 7.1 but for 7.0 as the OpenShift profile is probably not set you will not get the proper module name. Can you test with EAP 7.0 ?

Reply:
@jeffmaury you're right. It fails. But i think it must be part of https://issues.jboss.org/browse/JBIDE-22138?focusedCommentId=13608828&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13608828. @adietish wdyt?

Reply:
@bdshadow Yes, since the reason for the module name not being handled correctly lies in JBIDE-22138 aka upstream. As long as upstream isn't fixed the decision whether we want to apply this fix boils down to whether we want eap-70 or eap-71 templates to fail. I'd opt for sticking to the status quo aka not applying this fix.
@jeffmaury @bdshadow thoughts?